### DISCUSSION TITLE
data attribute JIT pseudo classes

### DISCUSSION THREAD

**Discussion by DanielBailey-web** [tailwindlabs/tailwindcss#7130] [Ideas]
It would be nice to be able to target data html attributes like is possible in normal css.

helpful link
https://css-tricks.com/a-complete-guide-to-data-attributes/

In normal css, you can do stuff like this
```css
[data-size="large"] {
  padding: 2rem;
  font-size: 125%;
}
```

Now that JIT is standard it seems like pseudo classes could be added as Arbitrary values, without increasing the package size by a ton. I don't think this would have been viable before the JIT compiler.

Something like:
```
[data-size="large"]:w-lg
```

This would be extremely useful for when libraries use html data attributes (and you don't have control over the internal state to apply normal classes) such as in radix-ui.

**Comment by BrianHung** [comment]
One application of this is integrating [radix](https://www.radix-ui.com/) with tailwind. Radix uses data attributes like `data-open` or `data-on` to customize css. Right now, to use tailwind with radix, you have to install a plugin https://www.npmjs.com/package/tailwindcss-radix and write `radix-state-open:style` as opposed to something like `[data-state="on"]`.
